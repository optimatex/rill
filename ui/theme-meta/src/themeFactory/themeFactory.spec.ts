import {
  getTheme,
  mockThemeValuesMap,
  mockThemeValuesMap_2,
  shapeThemeValues,
  themeValuesAfterSetting_2,
} from 'test';

import { themeFactory, themeExtenderFactory } from '.';

describe('themeFactory', () => {
  it('themeFactory() should return proper object', () => {
    // @ts-ignore
    const factory = themeFactory(shapeThemeValues, mockThemeValuesMap);
    const value = factory('dark');
    const desired = getTheme();

    expect(value).toEqual(desired);
  });

  it('themeExtenderFactory() with only valuesMap provided', () => {
    const theme = getTheme();
    // @ts-ignore
    const value = themeExtenderFactory(null, mockThemeValuesMap_2, theme);

    expect(value).toMatchSnapshot();
  });
});
