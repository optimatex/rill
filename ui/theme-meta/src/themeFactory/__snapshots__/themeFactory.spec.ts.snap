// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themeFactory themeExtenderFactory() with only valuesMap provided 1`] = `
Object {
  "button": Object {
    "background": "yellow",
    "color": "pink",
  },
  "buttonActive": Object {
    "background": "pink",
    "color": "grey",
  },
  "card": Object {
    "background": "blue",
    "color": "cyan",
  },
  "column": Object {
    "maxWidth": 560,
  },
  "font": Object {
    "fontFamily": "roboto",
    "fontSize": 8,
    "fontWeight": "bolder",
  },
  "frame": Object {
    "rowGap": 16,
    "sideIndent": 8,
    "verticalIndent": 5,
  },
  "layout": Object {
    "columnGap": 18,
    "rowGap": 18,
  },
  "mode": "dark",
  "surface": Object {
    "background": "orange",
    "color": "green",
  },
}
`;
